packages/hub/node-tests/@core/compiler-test.ts

```diff
@@ -484,6 +484,31 @@ if (process.env.COMPILER) {
         `);
       });
 
+      it('can compile schema class promise functions', async function () {
+        let { compiled } = await cards.load(`${realm}person`);
+        let source = getFileCache().getModule(compiled.schemaModule.global, 'browser');
+        expect(source).to.containsSource(`
+          then(resolve) {
+            let fields = ["lastName", "aboutMe", "fullName"];
+            return Promise.all(fields.map(field => this[field])).then(values => resolve(Object.fromEntries(fields.map((field, i) => [field, values[i]]))));
+          }
+        `);
+        expect(source).to.containsSource(`
+          catch(err) {
+            if (err) {
+              err();
+            }
+          }
+        `);
+        expect(source).to.containsSource(`
+          finally(cb) {
+            if (cb) {
+              cb();
+            }
+          }
+        `);
+      });
```

Comment:
Same here, it's not clear to me the intended use for this

Reply:
so when you want to use a finally for a promise you provide a callback. So a consumer of this promise that wants to use the finally would do this:

```js
promise.then(() => {
  // doSomething();
})
.finally(() => {
  doMyCustomTearDown();
});
```
so the finally needs to take as input a callback which is provided from outside the promise, `cb`. If finally is called with a callback, then we'll execute it.

Reply:
looking thru the docs on MDN, it looks like the `cb` param is required, so the `if` that I'm doing is unnecessary. and It is probably better to use the idea I was sketching out above with the `this.#allValues` getter that is returning the `Promise.all()` for getting all our values. if we leverage that, then we can have a finally that looks like:
```js
finally(cb) {
  this.#allValues.finally(cb);
}
```